src/cpu/aarch64/cpu_isa_traits.cpp

```diff
@@ -0,0 +1,148 @@
+/*******************************************************************************
+* Copyright 2019-2020 Intel Corporation
+* Copyright 2020 FUJITSU LIMITED
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include <cstring>
+#include <mutex>
+
+#include "common/utils.hpp"
+
+#include "cpu/aarch64/cpu_isa_traits.hpp"
+#include "cpu/cpu_engine.hpp"
+#include "cpu/platform.hpp"
+
+namespace dnnl {
+namespace impl {
+namespace cpu {
+namespace aarch64 {
+
+namespace {
+#ifdef DNNL_ENABLE_MAX_CPU_ISA
+
+set_before_first_get_setting_t<cpu_isa_t> &max_cpu_isa() {
+    static set_before_first_get_setting_t<cpu_isa_t> max_cpu_isa_setting;
+    return max_cpu_isa_setting;
+}
+
+bool init_max_cpu_isa() {
+    if (max_cpu_isa().initialized()) return false;
+
+    cpu_isa_t max_cpu_isa_val = isa_all;
+    char buf[64];
+    if (getenv("DNNL_MAX_CPU_ISA", buf, sizeof(buf)) > 0) {
+
+#define IF_HANDLE_CASE(cpu_isa) \
+    if (std::strcmp(buf, cpu_isa_traits<cpu_isa>::user_option_env) == 0) \
+    max_cpu_isa_val = cpu_isa
+#define ELSEIF_HANDLE_CASE(cpu_isa) else IF_HANDLE_CASE(cpu_isa)
+
+        IF_HANDLE_CASE(isa_all);
+        ELSEIF_HANDLE_CASE(asimd);
+        ELSEIF_HANDLE_CASE(sve_512);
+
+#undef IF_HANDLE_CASE
+#undef ELSEIF_HANDLE_CASE
+    }
+
+    return max_cpu_isa().set(max_cpu_isa_val);
+}
+#endif
+} // namespace
+
+struct isa_info_t {
+    isa_info_t(cpu_isa_t aisa) : isa(aisa) {};
+
+    // this converter is needed as code base defines certain ISAs
+    // that the library does not expose (eg sve),
+    // so the internal and external enum types do not coincide.
+    dnnl_cpu_isa_t convert_to_public_enum(void) const {
```

Comment:
Actually after a discussion with Evarist I changed my mind and I think we can leave it as is because ARM support is experimental and there is no users for this particular API. For sure this issue should be resolved in the future. Sorry for the noise.